domain/schema/testing/controllersuite.go

```diff
@@ -51,12 +51,13 @@ func (s *ControllerSuite) ControllerTxnRunner() coredatabase.TxnRunner {
 	return s.TxnRunner()
 }
 
-// SeedControllerUUID sets the uuid in the controller table to the default
-// testing value. It does not add any other controller config.
-func (s *ControllerSuite) SeedControllerUUID(c *gc.C) string {
-	controllerUUID := jujutesting.ControllerTag.Id()
+// SeedControllerTable sets the uuid in the controller table to the default
+// testing value and the controller mode uuid to the supplied value. It does not
+// add any other controller config.
+func (s *ControllerSuite) SeedControllerTable(c *gc.C, controllerModelUUID string) (controllerUUID string) {
```

Comment:
To ensure correctness, can we error out if the controllerModelUUID is empty.

Reply:
This is just for tests, I don't see why it should be a problem if its empty.

Reply:
What about if it defaults to the value mentioned in the previous comment if it is empty?

Reply:
Or I could even have two methods, SeedControllerUUID that uses a default value and this that uses the supplied value

Reply:
Having thought more, I think `controllerModelUUID` should be dropped from this, and we just generate a random one. We should have another function that allows you to set it. Then we can get rid of the empty string issue. Cleaner API.